### DISCUSSION TITLE
xrdp 0.9.22-3 doesnt contain libvnc.so anymore

### DISCUSSION THREAD

**Discussion by circip** [neutrinolabs/xrdp#2678] [General]
Hi, we are trying to install on RockyLinux 8.7 new xrdp package 0.9.22-3 and we discovered that , 
differently from previous 0.9.21-1 release , libvnc.so and other libraries are not present (see below report).
Is this an expected change and we need to install other dependencies?
Thanks in advance for the support!

_ rpm -qlp xrdp-0.9.22-3.el8.x86_64.rpm  | grep lib64
/usr/lib64/xrdp
/usr/lib64/xrdp/libcommon.so.0
/usr/lib64/xrdp/libcommon.so.0.0.0
/usr/lib64/xrdp/libscp.so.0
/usr/lib64/xrdp/libscp.so.0.0.0
/usr/lib64/xrdp/libxrdp.so.0
/usr/lib64/xrdp/libxrdp.so.0.0.0
/usr/lib64/xrdp/libxrdpapi.so.0
/usr/lib64/xrdp/libxrdpapi.so.0.0.0

 rpm -qlp xrdp-0.9.21-1.el8.x86_64.rpm  | grep lib64
/usr/lib64/librfxencode.so.0
/usr/lib64/librfxencode.so.0.0.0
/usr/lib64/xrdp
/usr/lib64/xrdp/libcommon.so.0
/usr/lib64/xrdp/libcommon.so.0.0.0
/usr/lib64/xrdp/libmc.so
/usr/lib64/xrdp/libscp.so.0
/usr/lib64/xrdp/libscp.so.0.0.0
**/usr/lib64/xrdp/libvnc.so**
/usr/lib64/xrdp/libxrdp.so.0
/usr/lib64/xrdp/libxrdp.so.0.0.0
/usr/lib64/xrdp/libxrdpapi.so.0
/usr/lib64/xrdp/libxrdpapi.so.0.0.0
/usr/lib64/xrdp/libxup.so_

**Comment by metalefty** [comment]
The issue is fixed in 0.9.22-4 package.

See also:
- https://bugzilla.redhat.com/show_bug.cgi?id=2207733
- https://src.fedoraproject.org/rpms/xrdp/c/812ea7eda56e04c4d58bdc5635c1140db60fa768?branch=rawhide

**Comment by matt335672** [comment]
Packages will appear first here:-

https://packages.fedoraproject.org/pkgs/xrdp/xrdp/

**Comment by circip** [comment]
Thanks a lot for the prompt feedbacks, any chance to get epel repo updated
soon ?

Best Regards


Il giorno mer 17 mag 2023 alle ore 11:07 matt335672 <
***@***.***> ha scritto:

**Comment by matt335672** [comment]
It's not our repo - it belongs to the Fedora guys.

I'm afraid you'll have to wait until your package appears in testing, or roll back to 0.9.21-1.

**Comment by circip** [comment]
Ok, thanks a lot , appreciated your interest.
BR


Il giorno mer 17 mag 2023 alle ore 15:28 matt335672 <
***@***.***> ha scritto:

**Comment by matt335672** [comment]
xrdp-0.9.22-4 is now in testing.

It has a significant issue :-
- #2687
- https://bugzilla.redhat.com/show_bug.cgi?id=2208248

**Comment by hossinasaadi** [comment]
i ran to this problem on centos 7 and how i solve the problem :
download your arch from https://koji.fedoraproject.org/koji/buildinfo?buildID=2201790
```
yum remove xrdp
wget https://kojipkgs.fedoraproject.org//packages/xrdp/0.9.22/4.el7/x86_64/xrdp-0.9.22-4.el7.x86_64.rpm
sudo rpm -i xrdp-0.9.22-4.el7.x86_64.rpm

```

**Reply by matt335672** [reply]
Thanks @hossinasaadi 

That's fine, but:-
1) Check my post above
2) Most users may need the `xrdp-selinux` package too. Using `yum` or `dnf` to install local packages will check this for you. Using `rpm` to do this isn't best practice.

**Reply by chernotic** [reply]
Thank you very much, I suffered for a week, but everything is so simple.

**Comment by abdeljalil09** [comment]
Hello guys this is how i fix the issue in rocky9

i have already a non working version of xrdp installed and this is what i did

` dnf install https://kojipkgs.fedoraproject.org//packages/xrdp/0.9.22/4.el9/x86_64/xrdp-0.9.22-4.el9.x86_64.rpm --allowerasing`

**Comment by matt335672** [comment]
0.9.22-5 is available from koji, and should appear in testing shortly.

**Comment by aroute** [comment]
The 0.9.22/4.el9 doesn't seem to solve this for me (Rocky 8.8). I added this entry as the error asks, to no avail. `add a valid entry like lib=libxrdp-vnc.so or similar` Any ideas?

**Reply by abdeljalil09** [reply]
replace el9 by el8 and it will work

**Comment by circip** [comment]
Thanks for the update. However still not possible not use yum to install
this package. I understood that fedorarepo are outside of this project, but
someone responsible should warn them to force an update, considering
several automated installations worldwide have been disrupted by .22.
BR

Il giorno ven 19 mag 2023 alle ore 15:47 matt335672 <
***@***.***> ha scritto:

**Comment by matt335672** [comment]
@circip - We made the EPEL project aware of the situation as soon as we could. They are aware of its importance and are following their own processes to get the update out. We can do no more from here.

@aroute - it's likely the update is not installed correctly. You shouldn't get that error with 0.9.22-4.

**Reply by circip** [reply]
@matt335672 , thanks for the feedback. However I don't think they have perceived the importance of this blocking issue. Still not updated epel repo at today. 3 weeks is not a fair reaction time , in my humble opinion.

**Comment by metalefty** [comment]
EPEL package has been updated, closing.
